src/MakePIAPortable/MakePIAPortable.csproj

```diff
@@ -26,35 +26,69 @@
+    <ILDAsmFileLocation Condition="$([MSBuild]::IsOSPlatform('osx'))">$(Pkgruntime_osx-x64_Microsoft_NETCore_ILDAsm)\runtimes\osx-x64\native\ildasm</ILDAsmFileLocation>
+    
+    <ILAsmFileLocation Condition="$([MSBuild]::IsOSPlatform('windows'))">$(Pkgruntime_win-x64_Microsoft_NETCore_ILAsm)\runtimes\win-x64\native\ilasm.exe</ILAsmFileLocation>
+    <ILAsmFileLocation Condition="$([MSBuild]::IsOSPlatform('linux'))">$(Pkgruntime_linux-x64_Microsoft_NETCore_ILAsm)\runtimes\linux-x64\native\ilasm</ILAsmFileLocation>
+    <ILAsmFileLocation Condition="$([MSBuild]::IsOSPlatform('osx'))">$(Pkgruntime_osx-x64_Microsoft_NETCore_ILAsm)\runtimes\osx-x64\native\ilasm</ILAsmFileLocation>
+    
+    <MakePIAPortableExe>MakePIAPortable.exe</MakePIAPortableExe>
+    <!-- Some machines might not have current directory in the PATH, force linux machines to run from current directory. -->
+    <MakePIAPortableExe Condition="$([MSBuild]::IsOsPlatform('linux'))">./MakePIAPortable.exe</MakePIAPortableExe>
+    <ILToolsExist Condition="Exists($(ILDAsmFileLocation)) AND Exists($(ILAsmFileLocation))">true</ILToolsExist>
+  </PropertyGroup>
   
-  <UsingTask TaskName="FindILDAsm" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
-    <ParameterGroup>
-      <ILDAsmFileLocation ParameterType="System.String" Output="true" />
-    </ParameterGroup>
-    <Task>
-      <Using Namespace="Microsoft.Build.Utilities" />
-      <Code Type="Fragment" Language="cs">
-        <![CDATA[
-          ILDAsmFileLocation = Microsoft.Build.Utilities.ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("ildasm.exe" , TargetDotNetFrameworkVersion.Latest);
-        ]]>
-      </Code>
-    </Task>
-  </UsingTask>
-  <Target Name="GeneratePortablePIA" BeforeTargets="SignFiles" AfterTargets="Build">
-    <FindILDAsm>
-      <Output PropertyName="ILDAsmFileLocation" TaskParameter="ILDAsmFileLocation" />
-    </FindILDAsm>
+  <Target Name="GeneratePortablePIA" BeforeTargets="SignFiles" AfterTargets="Build" Inputs="@(MakePIAPortableFiles)" Outputs="@(MakePIAPortableFiles->'$(PIAOutput)\%(Filename)%(Extension)')">
     <Message Importance="High" Text="ILDAsmFileLocation: $(ILDAsmFileLocation)" />
-    <Error Condition="'$(ILDAsmFileLocation)' == ''" Text="Unable to locate ILDasm.exe" />
-    <Exec Command="$(OutputPath)MakePortableLegacyPIAS.cmd &quot;$(ILDAsmFileLocation)&quot; &quot;$(PIAOutput)&quot;" />
+    <Message Importance="High" Text="ILAsmFileLocation: $(ILAsmFileLocation)" />
+    <Error Condition="'$(ILToolsExist)' != 'true'" Text="Unable to retrieve IL tools needed for MakePIAPortable" />
+
+    <!-- On linux, make MakePIAPortable exe runnable -->
+    <Exec Command="chmod 700 $(MakePIAPortableExe)" WorkingDirectory="$(OutputPath)" Condition="$([MSBuild]::IsOSPlatform('linux'))" />
```

Comment:
><Exec Command="chmod 700 $(MakePIAPortableExe)" WorkingDirectory="$(OutputPath)" Condition="$([MSBuild]::IsOSPlatform('linux'))" /> [](http://example.com/codeflow?start=4&length=131)

If you keep targetting net46:
* I think you want this condition for '!Windows' rather than 'Linux'
* Would it really be better to explicitly run it under mono?